libfwupdplugin/fu-device.c

```diff
@@ -2972,6 +2972,63 @@ fu_device_probe_invalidate (FuDevice *self)
 	priv->done_setup = FALSE;
 }
 
+/**
+ * fu_device_report_metadata_pre:
+ * @self: A #FuDevice
+ *
+ * Collects metadata that would be useful for debugging a failed update report.
+ *
+ * Returns: (transfer full): A #GHashTable
+ *
+ * Since: 1.5.0
+ **/
+GHashTable *
+fu_device_report_metadata_pre (FuDevice *self)
+{
+	FuDeviceClass *klass = FU_DEVICE_GET_CLASS (self);
+	g_autoptr(GHashTable) metadata = NULL;
+
+	g_return_val_if_fail (FU_IS_DEVICE (self), NULL);
+
+	/* metadata for all devices */
+	metadata = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
+	g_hash_table_insert (metadata,
+			     g_strdup ("GType"),
+			     g_strdup (G_OBJECT_TYPE_NAME (self)));
```

Comment:
Is this actually valuable for all the devices?

Reply:
For things like VLI or Wacom that are subclassed quite deeply, yes. For others perhaps less so.

Reply:
But so in those it actually matters, shouldn't they be device metadata made by the plugin?

Reply:
Sure, I can do that if you think it's clearer.

Reply:
Yes I do think it's clearer.

Reply:
Okay, fixed.